### PR TITLE
Fix race condition is using CreateRequestQueue for inbound chunks

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

If two threads try to create a chunk at the same time, after https://github.com/ray-project/ray/pull/16523 there was a possibility one of them would fail and abort the chunk receive.

This results in the pull request getting retried at a higher level, leading to delays.

### CODE REVIEW

File: src/ray/object_manager/object_buffer_pool.cc [hunk 1]
```diff
@@ -117,24 +117,28 @@ std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> ObjectBufferPool::Cr
         plasma::flatbuf::ObjectSource::ReceivedFromRemoteRaylet);
     lock.lock();
```
Comment (ericl) [inline src/ray/object_manager/object_buffer_pool.cc:118]: The race condition is introduced by releasing the lock above. We need to double-check the condition upon re-entering the block.

**Comment by rkooo567** [comment]
You think this issue could be related? https://discuss.ray.io/t/big-cluster-job-failing-due-to-sigbus-in-plasma/2578/9

**Review by rkooo567 — APPROVED** [review]

File: src/ray/object_manager/object_buffer_pool.cc [hunk 2]
```diff
@@ -117,24 +117,28 @@ std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> ObjectBufferPool::Cr
...
+      std::vector<boost::asio::mutable_buffer> buffer;
+      if (!s.ok()) {
+        // Create failed. The object may already exist locally. If something else went
+        // wrong, another chunk will succeed in creating the buffer, and this
+        // chunk will eventually make it here via pull requests.
+        return std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status>(
+            errored_chunk_, ray::Status::IOError(s.message()));
+      }
+      // Read object into store.
+      uint8_t *mutable_data = data->Data();
```
Comment (stephanie-wang) [inline src/ray/object_manager/object_buffer_pool.cc:132]: Shouldn't the copy take place outside of the `if` block still?

**Comment by ericl** [comment]
@rkooo567 I don't think I can say. There are a number of issues with memory management that won't be addressed until https://github.com/ray-project/ray/pull/16394 is finished (and tested).

File: src/ray/object_manager/object_buffer_pool.cc (hunk 2 above)
Comment (ericl) [reply src/ray/object_manager/object_buffer_pool.cc:132]: Yeah good point. Let me look into this more, I think this function is quite brittle and we could rewrite it to be more resilient to duplicate chunk puts.

Comment (ericl) [reply src/ray/object_manager/object_buffer_pool.cc:132]: Turns out this was only to initialize the buffer (the actual copy is done in ObjectManager).

**Pull request merged**